src/source/Ice/IceAgent.c

```diff
@@ -766,18 +768,19 @@ STATUS iceAgentShutdown(PIceAgent pIceAgent)
     CHK(pIceAgent != NULL, STATUS_NULL_ARG);
     CHK(!ATOMIC_EXCHANGE_BOOL(&pIceAgent->shutdown, TRUE), retStatus);
 
+    // Shutdown all the timers during iceAgentShutdown; do not partially shut down timers.
     if (pIceAgent->iceAgentStateTimerTask != MAX_UINT32) {
-        CHK_STATUS(timerQueueCancelTimer(pIceAgent->timerQueueHandle, pIceAgent->iceAgentStateTimerTask, (UINT64) pIceAgent));
+        CHK_LOG_ERR(timerQueueCancelTimer(pIceAgent->timerQueueHandle, pIceAgent->iceAgentStateTimerTask, (UINT64) pIceAgent));
```

Comment:
This is good so we do not abort trying to cancel the other timers.  However you should also *set* the `retStatus` so the overall function can still return the ERROR status to the caller to notify it that the `iceAgentShutdown` operation did not fully succeed.

Perhaps something like this:

```
retStatus = timerQueueCancelTimer(pIceAgent->timerQueueHandle, pIceAgent->iceAgentStateTimerTask, (UINT64) pIceAgent));
CHK_LOG_ERR(retStatus)
```

Note that `CHK_LOG_ERR` is a no-op if the retStatus passed to it is a STATUS_SUCCESS.

Reply:
One adjustment to that suggestion is that we'll also need to not set the status back to a success after something took too long, so an additional check will be needed.